src/feed/feed.service.ts

```diff
@@ -680,6 +680,26 @@ export class FeedService {
         return updatedFeedUserIds;
     }
 
+    /**
+     * Remove a post from discovery feeds
+     *
+     * @param post Post to remove from discovery feeds
+     * @returns IDs of the topics that had their discovery feed updated
+     */
+    async removePostFromDiscoveryFeeds(post: Post) {
+        // Work out which topics have the post in their discovery feeds
+        const updatedTopicIds = (
+            await this.db('discovery_feeds')
+                .where('post_id', post.id)
+                .select('topic_id')
+        ).map((topic: { topic_id: number }) => topic.topic_id);
+
+        // Remove the post from the discovery feeds
+        await this.db('discovery_feeds').where('post_id', post.id).delete();
+
+        return updatedTopicIds;
```

Comment:
Besides for testing purposes, is there value is computing and returning `updatedTopicIds`? We can test it without it by looking at the discovery feed content before and after, so maybe good to skip the extra database query here?

Reply:
Yeh this is redundant, good catch 🎣 

Fixed in https://github.com/TryGhost/ActivityPub/pull/1418/commits/467346d41756836979010cf07efe36d29536d488